Changed Lots of core files

Lamrsw opened the pull request.

Lamrsw committed:
Boat changes and images

Added getters and setters to boat class and changed boat image

Lamrsw committed:
Added stuff

Lamrsw committed:
Boat extends rectangle

Also collisions

Lamrsw committed:
Boat

Lamrsw committed:
Added Game Over

Lamrsw committed:
Added random textures to obstacles

Lamrsw committed:
Aiboat

JL01-DevOps reviewed (approved):
Go ahead, great work.

Lamrsw committed:
CPU boats and barriers

Added rudementary random movement to ai boats and added barriers between the lanes
